app/src/main/java/edu/wpi/first/shuffleboard/app/MainWindowController.java

```diff
@@ -85,48 +97,27 @@ private void initialize() throws IOException {
             Recorder.getInstance().runningProperty(),
             running -> running ? "Stop recording" : "Start recording"));
     FxUtils.bind(root.getStylesheets(), stylesheets);
-    NetworkTableTreeWidget networkTables = new NetworkTableTreeWidget();
-    networkTables.getTree().getSelectionModel().selectedItemProperty().addListener((__, prev, cur) -> {
-      if (cur == null) {
-        selectedEntry = null;
-      } else {
-        selectedEntry = cur.getValue();
-      }
-    });
-    networkTables.getTree().setRowFactory(view -> {
-      TreeTableRow<NetworkTableEntry> row = new TreeTableRow<>();
-      row.hoverProperty().addListener((__, wasHover, isHover) -> {
-        if (!row.isEmpty()) {
-          setHighlightedWithChildren(row.getTreeItem(), isHover);
-        }
-      });
-      makeSourceRowDraggable(row);
-      return row;
-    });
-    networkTables.getTree().setOnContextMenuRequested(e -> {
-      TreeItem<NetworkTableEntry> selectedItem = networkTables.getTree().getSelectionModel().getSelectedItem();
-      if (selectedItem == null) {
-        return;
-      }
 
-      DataSource<?> source = selectedItem.getValue().get();
-      List<String> widgetNames = Widgets.widgetNamesForSource(source);
-      if (widgetNames.isEmpty()) {
-        // No known widgets that can show this data
-        return;
+    PluginLoader.getDefault().getKnownPlugins().addListener((ListChangeListener<Plugin>) c -> {
+      while (c.next()) {
+        if (c.wasAdded()) {
+          c.getAddedSubList().forEach(plugin -> {
+            plugin.loadedProperty().addListener((__, was, is) -> {
+              if (is) {
+                System.out.println("GENERATING FOR " + plugin.getName());
+                setup(plugin);
+              } else {
+                System.out.println("REMOVING ITEMS FOR " + plugin.getName());
```

Comment:
I left those in while I was debugging. We use Java's builtin `Logger` class for logging; this was a quick and dirty (and easy to read) way to debug